### PR TITLE
Fixes Search in the search editor does not respect the modified sort field

### LINKED ISSUE #108286: Search in the search editor windows does not respect the modified sort field
- VSCode Version: 1.50.0-insider
- OS Version: macOS Catalina

Steps to Reproduce:

In the same vein as  #103625, but with ```modified``` selected (works with countDescending).

1. Settings > "Search: Sort Order" > select "modified"
2. Ctrl+P "> Search editor" (open the search editor)
3. Search something, results appear sorted by path

Does this issue occur when all extensions are disabled?: Yes

/cc @JacksonKearl @damienmg <3

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #108286

### CODE REVIEW

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorActions.ts [hunk 1]
```diff
@@ -178,11 +179,17 @@ export const createEditorFromSearchResult =
...
-		const sortOrder = configurationService.getValue<ISearchConfigurationProperties>('search').sortOrder;
-
+		const fileService = accessor.get(IFileService);
 
 		telemetryService.publicLog2('searchEditor/createEditorFromSearchResult');
 
+		const sortOrder = configurationService.getValue<ISearchConfigurationProperties>('search').sortOrder;
+		if (sortOrder === SearchSortOrder.Modified) {
+			// Ensure all matches have retrieved their file stat
+			const files = searchResult.matches().filter(f => !f.fileStat).map(f => f.resolveFileStat(fileService));
```
Comment (JacksonKearl) [inline src/vs/workbench/contrib/searchEditor/browser/searchEditorActions.ts:189]: Did you find a case where this was needed? I'd expect the stats to all be resolved based from the existing search.

**Comment by JacksonKearl** [comment]
Thanks for the PR! I think this could be cleaner by moving the logic for statting the files into the SearchResult class. As-is we're duplicating the same logic 3 times over.

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorActions.ts (hunk 1 above)
Comment (jeanp413) [reply src/vs/workbench/contrib/searchEditor/browser/searchEditorActions.ts:189]: yeah you are right, I'll revert this changes

**Comment by jeanp413** [comment]
Reverted the unneeded changes, do you still think I should move the statting logic to `SearchResult`? Not sure about it since we still need a version that accepts a subset of the matches
https://github.com/microsoft/vscode/blob/c0d465ce1c2c1992b30bb4219ed17d88529c87f2/src/vs/workbench/contrib/search/browser/searchView.ts#L1890-L1893

**Review by JacksonKearl — APPROVED** [review]
> Reverted the unneeded changes, do you still think I should move the statting logic to SearchResult?

You're right I dont think we need to anymore.

Looks good to me, thanks for this!

**Pull request merged**